app/models/task.py

```diff
@@ -20,19 +20,23 @@ def to_dict(self, include_completed_at=False, include_goal_id=False):
             "id": self.id,
             "title": self.title,
             "description": self.description,
-            "is_complete": bool(self.completed_at),
-            "goal_id": self.goal.id if self.goal else None,  # Default to None
+            "is_complete": self.completed_at is not None,
+            # "goal_id": self.goal.id if self.goal else None,  # Default to None
             # "goal_title": self.goal.title if self.goal else None  # Default to None
 
         }
-        if include_completed_at:
-            task_dict["completed_at"] = self.completed_at
-
-        if include_goal_id and self.goal:
-            task_dict["goal_id"] = self.goal.id
-            task_dict["goal_title"] = self.goal.title
+        if include_goal_id and self.goal_id:
```

Comment:
This works as written (with you explicitly passing in `True` for `include_goal_id` in a few places). But the intended behavior was to conditionally include the goal id in any Task output if it belongs to a goal, with no need to "turn it on" in certain routes.

The reason the author might feel like this is OK for goal info, but not for task info (on goals) is that the goal info is a single value, while the task info is a whole additional collection of unknown size.

Personally, I disagree with the `goal_id` coming and going (I would have always included the `goal_id` with a `None` value if it didn't belong to a Goal rather than modifying the "shape" of a Task response), but hey, a spec is a spec. 🤷